crates/client/eth/src/l1_messaging.rs

```diff
@@ -0,0 +1,526 @@
+use anyhow::Context;
+use futures::StreamExt;
+use std::sync::Arc;
+
+use crate::client::StarknetCoreContract::LogMessageToL2;
+use crate::client::{EthereumClient, StarknetCoreContract};
+use crate::utils::u256_to_felt;
+use alloy::primitives::{keccak256, FixedBytes, U256};
+use alloy::sol_types::SolValue;
+use blockifier::transaction::transactions::L1HandlerTransaction as BlockifierL1HandlerTransaction;
+use dc_db::{l1_db::LastSyncedEventBlock, DeoxysBackend};
+use dp_utils::channel_wait_or_graceful_shutdown;
+use starknet_api::core::{ChainId, ContractAddress, EntryPointSelector, Nonce};
+use starknet_api::transaction::{
+    Calldata, Fee, L1HandlerTransaction, Transaction, TransactionHash, TransactionVersion,
+};
+use starknet_api::transaction_hash::get_transaction_hash;
+use starknet_types_core::felt::Felt;
+
+impl EthereumClient {
```

Comment:
nitpick: i was thinking we can pass the `EthereumClient` as a ref to this function instead of implementing it inside it. i was thinking of the `EthereumClient` as a more generalised struct but that's just my view :)

Reply:
I see your point. For me EthereumClient is more of an interface for interacting with L1 (and core contract). wdyt @cchudant ?

Reply:
i don't have a preference, but i would think that anvil already provides us the generalized ethereum client, that struct is more of the interface to the core contract more specifically
maybe it should be renamed core contract client then

Reply:
let's keep it like this for now I think, not sure it's really core contract specific